src/Tribe/iCal.php

```diff
@@ -237,77 +321,287 @@ private function get_month_view_events() {
 	}
 
 	/**
-	 * Generates the iCal file
+	 * Set the headers before the file is delivered.
 	 *
-	 * @param int|null $post If you want the ical file for a single event
-	 * @param boolean $echo Whether the content should be echoed or returned
+	 * @since TBD
 	 */
-	public function generate_ical_feed( $post = null, $echo = true ) {
-		$tec         = Tribe__Events__Main::instance();
-		$events      = '';
-		$blogHome    = get_bloginfo( 'url' );
-		$blogName    = get_bloginfo( 'name' );
+	protected function set_headers() {
+		header( 'HTTP/1.0 200 OK', true, 200 );
+		header( 'Content-type: text/calendar; charset=UTF-8' );
+		header(
+			'Content-Disposition: attachment; filename="' . $this->get_file_name() . '"'
+		);
+	}
 
-		if ( $post ) {
-			$events_posts = is_array( $post ) ? $post : array( $post );
-		} elseif ( tribe_is_month() ) {
-			$events_posts = self::get_month_view_events();
-		} elseif ( tribe_is_organizer() ) {
-			$events_posts = $this->get_events_list( array(
-				'organizer'    => get_the_ID(),
-				'eventDisplay' => 'list',
-			) );
-		} elseif ( tribe_is_venue() ) {
-			$events_posts = $this->get_events_list( array(
-				'venue'        => get_the_ID(),
-				'eventDisplay' => tribe_get_request_var( 'tribe_event_display', 'list' ),
-			) );
-		} else {
+	/**
+	 * Get the file name of the *.ics file
+	 *
+	 * @since TBD
+	 *
+	 * @return mixed The calendar name
+	 */
+	protected function get_file_name() {
+		$event_ids = wp_list_pluck( $this->events, 'ID' );
+		$site = sanitize_title( get_bloginfo( 'name' ) );
+		$hash = substr( md5( implode( $event_ids ) ), 0, 11 );
+		$filename = sprintf( '%s-%s.ics', $site, $hash );
 
-			if ( ! $wp_query = tribe_get_global_query_object() ) {
-				return;
+		/**
+		 * Modifies the filename provided in the Content-Disposition header for iCal feeds.
+		 *
+		 * @var string       $filename
+		 * @var WP_Post|null $post
+		 */
```

Comment:
Let's add a `@since TBD` here.

Reply:
Oh this one is an existing filter that was moved around due the changes on the code, do you think is still relevant to add a `@since` even was not added on this PR?

Reply:
Perfect!